sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/workflowentity/WorkflowEntity.java

```diff
@@ -258,8 +269,16 @@ protected final Effect.Builder<S> effects() {
 
   public static class Workflow<S> {
 
-    final private List<Step> steps = new ArrayList<Step>();
+    final private List<Step> steps = new ArrayList<>();
+    final private List<StepConfig> stepConfigs = new ArrayList<>();
     final private Set<String> uniqueNames = new HashSet<>();
+    private Optional<Duration> workflowTimeout = Optional.empty();
+    private Optional<String> failoverToStepName = Optional.empty();
+    private Optional<Object> failoverToStepInput = Optional.empty();
+    private Optional<MaxRetries> failoverToRecoverStrategy = Optional.empty();
```

Comment:
`failoverMaxRetries`? 

btw, now that I read this once more, I think it's confusing to prefix those fields with `failoverTo`. It makes me think of a conversion from `failover to stepName`, for example. 

Maybe better to rename them all: `failoverStepName`, `failoverStepInput` and `failoverMaxRetries`. 

The method to declare the failover can stay as `failoverTo` as it aligns well with `transitionTo` and is the user facing API.

Reply:
makes sense, changing